src/libcore/fmt/mod.rs

```diff
@@ -81,6 +81,21 @@ pub type Result = result::Result<(), Error>;
 /// This type does not support transmission of an error other than that an error
 /// occurred. Any extra information must be arranged to be transmitted through
 /// some other means.
+///
+/// The key thing to remember here is that the type `fmt::Error` should not be
+/// confused with `std::error::Error`, which you may also have in scope.
+///
+/// # Example
+///
+/// ```rust
+/// use std::fmt;
+///
+/// let mut output = String::new();
+/// match fmt::write(&mut output, format_args!("Hello {}!", "world")) {
+///     Err(fmt::Error) => panic!("An error occurred"),
+///     _ => (),
+/// }
```

Comment:
I agree it would be more idiomatic, but that would not demonstrate the usage of `fmt::Error` - so I'm not sure if it would be worth including the example in that case.